### ISSUE TITLE
MySQL support for 'bit' data type 

### ISSUE THREAD

**Issue by yonatanmn** [typeorm/typeorm#3071]
**Issue type:**
[X] feature request

**Database system/driver:**
[X] `mysql` / `mariadb`

**TypeORM version:**
[X] `latest`

I'm working with existing DB, that was created with hibernate ORM. All booleans were created with `bit` as it is recommended with new versions of MySQL (not sure about it though)

Now I'm adding nestjs server with typeorm, and when I try to connect to the DB, it throws that `bit` data type is not supported by mysql driver. 
If I change it to tinyint, (or not specifying anything) it alters the table, and then the hibernate server is not working. 

What can I do? why not supporting 'bit'? 

Thanks

**Comment by frozenex** [comment]
+1

**Comment by vlapo** [comment]
I think `bit` should be represented as string (same as in pgsql).

Also we should also add `bool` and also `boolean` to mysql driver as they are synonyms for `TINYINT(1)`

https://dev.mysql.com/doc/refman/5.5/en/numeric-type-overview.html

**Comment by pleerock** [comment]
Yeah I think we should add those new data types. Its easy task to do actually.
